### PR TITLE
add new errors for Disks with timeouts

### PR DESCRIPTION
**Pull request by harshavardhana**
## Description
add new errors for Disks with timeouts

## Motivation and Context
for more debugging, available via `mc admin info --json` 
needs this PR https://github.com/minio/mc/pull/4649

## How to test this PR?
Just trigger disk timeouts you may see the errors

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Unit tests added/updated
- [ ] Internal documentation updated
- [ ] Create a documentation update request [here](https://github.com/minio/docs/issues/new?label=doc-change,title=Doc+Updated+Needed+For+PR+github.com%2fminio%2fminio%2fpull%2fNNNNN)

### CODE REVIEW

File: cmd/xl-storage-disk-id-check.go [hunk 1]
```diff
@@ -687,6 +687,9 @@ func (p *xlStorageDiskIDCheck) updateStorageMetrics(s storageMetric, paths ...st
 			if errors.Is(err, context.DeadlineExceeded) || errors.Is(err, context.Canceled) {
 				atomic.AddUint64(&p.totalErrsTimeout, 1)
 			}
+			if (err.Error() == context.DeadlineExceeded.Error()) || (err.Error() == context.Canceled.Error()) {
```
Comment (vadmeste) [inline cmd/xl-storage-disk-id-check.go]: We will be double counting the metrics here

**Review by vadmeste — COMMENTED** [review]
One comment

File: cmd/xl-storage-disk-id-check.go (hunk 1 above)
Comment (harshavardhana) [reply cmd/xl-storage-disk-id-check.go]: yes testing code, now removed @vadmeste

**Review by vadmeste — APPROVED** [review]
LGTM

**Pull request merged**